[KB-253] Description of kernel pca (#158)
chhoumann/thesis-chemcam: report_thesis/src/sections/background.tex

where $\mathbf{V}_n$ is the matrix that only contains the top $n$ eigenvectors.

\subsubsection{Kernel PCA}

> We provide a description of the \gls{kernel-pca} algorithm based on \citet{learningwithkernels}.

Reviewer: ```suggestion
We provide a brief overview of the \gls{kernel-pca} algorithm based on \citet{learningwithkernels}.
```

\gls{kernel-pca} is an extension of traditional \gls{pca} designed to handle nonlinear relationships among data points.
The core idea behind \gls{kernel-pca} is to map data into a higher-dimensional space using a kernel function, a technique known as the \textit{kernel trick}.
This mapping enables linear separation of data points in the higher-dimensional space, even if they are not linearly separable in the original space.

> The key idea of \gls{kernel-pca} is to extract the principal components of the data similar to \gls{pca}, as described in Section~\ref{subsec:pca}.

Reviewer: ```suggestion
The main idea behind \gls{kernel-pca} is to extract the principal components of the data, similar to \gls{pca} as described in Section~\ref{subsec:pca}.
```

where $\mathbf{V}_n$ is the matrix that only contains the top $n$ eigenvectors.

\subsubsection{Kernel PCA}

> The description of the \gls{kernel-pca} algorithm is based on \citet{learningwithkernels}.

Reviewer: ```suggestion
We provide a description of the \gls{kernel-pca} algorithm based on \citet{learningwithkernels}.
```

> \gls{kernel-pca} is an extension of the traditional \gls{pca} that is used to handle nonlinear relationships among data points.

Reviewer: ```suggestion
\gls{kernel-pca} is an extension of traditional \gls{pca} designed to handle nonlinear relationships among data points.
```

> The core idea behind \gls{kernel-pca} is to map the data into a higher-dimensional space using a kernel function, known as the \textit{kernel trick}. 

Reviewer: ```suggestion
The core idea behind \gls{kernel-pca} is to map data into a higher-dimensional space using a kernel function, a technique known as the \textit{kernel trick}.
```

> This mapping allows for the linear separation of data points in the higher-dimensional space, even when the data is not linearly separable in the original space.

Reviewer: ```suggestion
This mapping enables linear separation of data points in the higher-dimensional space, even if they are not linearly separable in the original space.
```

The core element of \gls{kernel-pca} is to obtain the covariance matrix in the higher-dimensional space. 
Mathematically, this is given by:
$$
	\mathbf{C} = \frac{1}{m} \sum_{j=1}^{m} \Phi(\mathbf{x}_j) \Phi(\mathbf{x}_j)^\top
$$

> where $m$ is the number of data points and $\Phi(\mathbf{x}_j)$ represents the mapping of the data point $\mathbf{x}_j$ into the higher-dimensional feature space. 

Reviewer: ```suggestion
where $m$ is the number of data points and $\Phi(\mathbf{x}_j)$ represents the mapping of data point $\mathbf{x}_j$ into the higher-dimensional feature space.
```

> where $m$ is the number of data points and $\Phi(\mathbf{x}_j)$ represents the mapping of the data point $\mathbf{x}_j$ into the higher-dimensional feature space. 

Reviewer: I believe this entire part is wrong (at least based on my understanding and what you write afterwards). In traditional PCA, the covariance matrix captures the variance and covariance between every pair of features in the dataset, which is crucial for identifying the principal components that capture the most variance in the data. But, as you say, this is not feasible in practice for large datasets - hence the kernel trick.

The term $\frac{1}{m}$ normalizes the sum, ensuring the covariance matrix represents the average pairwise relationships between data points.

> However, in practice, computing the covariance matrix is typically not feasible for large datasets, as computing the mapping function $\Phi(\mathbf{x}_j$ for each data point is expensive and often impractical due to the high dimensionality of the feature space. 

Reviewer: ```suggestion
However, computing this covariance matrix directly is often infeasible for large datasets due to the high dimensionality of the feature space. 
```
If you decide to omit/rewrite the above.

> Instead, Kernel PCA leverages the kernel trick, which allows us to compute the similarity between data points directly in the original space using a kernel function $k(\mathbf{x}_i, \mathbf{x}_j)$. 

Reviewer: ```suggestion
Kernel PCA addresses this by leveraging the kernel trick, which allows the computation of similarities between data points directly in the original space using a kernel function \(k(\mathbf{x}_i, \mathbf{x}_j)\). 
```

This kernel function implicitly computes the dot product $\Phi(\mathbf{x}_i)^\top \Phi(\mathbf{x}_j)$ in the higher-dimensional feature space without explicitly performing the mapping. 
By constructing the kernel matrix \(\mathbf{K}\) using these pairwise similarities, Kernel PCA can perform eigenvalue decomposition to obtain the principal components in the feature space, in the same fashion as regular \gls{pca} described in Section~\ref{subsec:pca}.

> However, the eigenvalue decomposition is performed on the kernel matrix $\mathbf{K}$ rather than the covariance matrix $\mathbf{C}$.

Reviewer: ```suggestion
However, in Kernel PCA, the eigenvalue decomposition is performed on the kernel matrix \(\mathbf{K}\) rather than the covariance matrix \(\mathbf{C}\).
```
